File: Ahorn/entities/CustomCliffSideWindFlag.jl
  
module FemtoHelperCustomCliffsideFlag

using ..Ahorn, Maple

@mapdef Entity "FemtoHelper/CustomCliffsideWindFlag" CustomCliffsideFlag(x::Integer, y::Integer, index::Integer=0, spritesPath::String="scenery/cliffside/flag", sineFrequency::Number=1.0, sineAmplitude::Number=1.0, naturalDraft::Number=0.0)

const placements = Ahorn.PlacementDict(
    "Custom Wind Flag (Femto Helper)" => Ahorn.EntityPlacement(
        CustomCliffsideFlag,
    )
)

Ahorn.editingOptions(entity::CustomCliffsideFlag) = Dict{String, Any}(
)

function flagSprite(entity::CustomCliffsideFlag)
    index = Int(get(entity.data, "index", 0))
    path = get(entity.data, "spritesPath", "scenery/cliffside/flag")
    lookup = lpad(string(index), 2, "0")

    return "$(path)$(lookup)"
end

function Ahorn.selection(entity::CustomCliffsideFlag)
    x, y = Ahorn.position(entity)
    sprite = flagSprite(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y, jx=0.0, jy=0.0)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::CustomCliffsideFlag, room::Maple.Room)
    sprite = flagSprite(entity)
    Ahorn.drawSprite(ctx, sprite, 0, 0, jx=0.0, jy=0.0)
end

end

File: Loenn/entities/CustomCliffSideWindFlag.lua
local FemtoHelperCustomCliffsideFlag = {}

FemtoHelperCustomCliffsideFlag.name = "FemtoHelper/CustomCliffsideWindFlag"
FemtoHelperCustomCliffsideFlag.depth = 8998
FemtoHelperCustomCliffsideFlag.justification = {0.5, 0.5}
FemtoHelperCustomCliffsideFlag.fieldInformation = {
    index = {
        fieldType = "integer",
        options = {
            0, 1, 2, 3,
            4, 5, 6, 7,
            8, 9, 10
        },
        editable = true
    }
}
FemtoHelperCustomCliffsideFlag.placements = {
    name = "default",
    data = {
        index = 0,
        spritesPath = "scenery/cliffside/flag",
        sineFrequency = 1,
        sineAmplitude = 1,
        naturalDraft = 0,
        depth = 8999,
    }
}

function FemtoHelperCustomCliffsideFlag.texture(room, entity)
    return entity.spritesPath .. "0" .. entity.index
end

return FemtoHelperCustomCliffsideFlag